File: Programmers/기능개발.js
const progresses = [93, 30, 55];
const speeds = [1, 30, 5];

function solution(progresses, speeds) {
    let answer = [];
    let temp = [];
    let comp;
    let count = 0;
    
    temp = progresses.map((x, i) => Math.ceil((100 - x) / speeds[i]));
    console.log("계산용 :", temp);

    if(temp.length == 1) {
        answer.push(1);
    }
    else {
        comp = temp[0];
        for(let i = 0; i < temp.length; i++) {
            if(temp[i] <= comp) {
                count++;
            } else {
                answer.push(count);
                count = 1;
                comp = temp[i];
            }
        }
        answer.push(count);
    }
    
    return answer;
}

const answer = solution(progresses, speeds);
console.log("답 :", answer);

File: Programmers/기능개발.py
from collections import deque
import math


def solution(progresses, speeds):

    global count
    temp = deque(list(map(lambda progress, speed: math.ceil((100 - progress) / speed), progresses, speeds)))
    answer = []
    print(temp)
    while temp:
        p = temp.popleft()
        count = 1

        while temp:

            if p >= temp[0]:
                count += 1
                temp.popleft()
            else:
                answer.append(count)
                break

    answer.append(count)

    return answer


print(solution(PROGRESSES, SPEEDS))
